freeipa/src/main/java/org/apache/directory/shared/kerberos/components/EncryptionKey.java

```diff
@@ -0,0 +1,330 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
```

Comment:
So what is the apache license? Can we copy the code and use it/modify it as long as we have the copyright statement?

Reply:
I think I answered my own question by doing a git grep on thunderhead git repo. I think it's fine.

Reply:
Cool. I think all we need to do is just include the original license, and if we've modified the code we just put some Cloudera modification header.